### PR TITLE
RavenDB-20110 - Import or restore of unsupported feature should be sk…

### PR DESCRIPTION
**Pull request by shiranshalom**
…ipped and warned about

### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-20110/Import-or-restore-of-unsupported-feature-should-be-skipped-and-warned-about

### Type of change

- Bug fix

### How risky is the change?

- Low 

### Backward compatibility

- Non breaking change

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing by Contributor

- Tests have been added that prove the fix is effective or that the feature works

### Testing by RavenDB QA team

- No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: src/Raven.Server/Smuggler/Documents/MultiShardedDestination.cs [hunk 1]
```diff
@@ -44,6 +44,7 @@ public MultiShardedDestination([NotNull] ISmugglerSource source, [NotNull] Shard
 
         public async ValueTask<IAsyncDisposable> InitializeAsync(DatabaseSmugglerOptionsServerSide options, SmugglerResult result, long buildVersion)
         {
+            options.OperateOnDatabaseRecordTypes &= ~(DatabaseRecordItemType.HubPullReplications | DatabaseRecordItemType.SinkPullReplications);
```
Comment (ppekrol) [inline src/Raven.Server/Smuggler/Documents/MultiShardedDestination.cs]: Better to move those flags to DatabaseSmugglerOptions as internal and call them `ShardingNotSupportedDatabaseSmugglerOptions` or sth like that.

Comment (shiranshalom) [reply src/Raven.Server/Smuggler/Documents/MultiShardedDestination.cs]: Done

**Comment by ppekrol** [comment]
@shiranshalom can you check `SlowTests.Sharding.Backup.ShardedSmugglerTests.CanExportAndImportDatabaseRecordToShardDatabase` test? Looks related :)

**Comment by shiranshalom** [comment]
> @shiranshalom can you check `SlowTests.Sharding.Backup.ShardedSmugglerTests.CanExportAndImportDatabaseRecordToShardDatabase` test? Looks related :)

done :)

**Comment by karmeli87** [comment]
I wonder if we should take that one step further, because probably people will migrate to sharding by importing.
So we need to check also this list https://ravendb.net/docs/article-page/6.0/csharp/sharding/unsupported
And skip/warn and check the behavior for each point there that affected by import (indexes, backup, etc.. )

for instance
- import an index with output reduce to collection.
- snapshot backup
- concurrent subscriptions
etc..

File: src/Raven.Server/Smuggler/Documents/ShardedDatabaseSmuggler.cs [hunk 2]
```diff
@@ -50,6 +51,63 @@ protected override async Task<SmugglerProgressBase.DatabaseRecordProgress> Proce
...
+        }
+
+        protected override bool ShouldSkipIndex(IndexDefinitionAndType index, out string msg)
+        {
+            msg = null;
+            var definition = (IndexDefinition)index.IndexDefinition;
+
+            if (index.Type is IndexType.MapReduce or IndexType.AutoMapReduce)
+            {
+                msg = $"Skipped index '{definition.Name}'. Map-Reduce is currently not supported in Sharding";
```
Comment (karmeli87) [inline src/Raven.Server/Smuggler/Documents/ShardedDatabaseSmuggler.cs]: only indexes that output the reduce result to a collection should be skipped.

**Review by karmeli87 — CHANGES_REQUESTED** [review]
Please also add a test for the output to collection index

File: src/Raven.Server/Smuggler/Documents/ShardedDatabaseSmuggler.cs (hunk 2 above)
Comment (shiranshalom) [reply src/Raven.Server/Smuggler/Documents/ShardedDatabaseSmuggler.cs]: done

**Comment by shiranshalom** [comment]
> Please also add a test for the output to collection index

done

**Review by karmeli87 — APPROVED** [review]

**Review by ppekrol — APPROVED** [review]
LGTM, but please check failing tests

**Comment by shiranshalom** [comment]
> LGTM, but please check failing tests

done :)

File: src/Raven.Server/Smuggler/Documents/ShardedDatabaseSmuggler.cs [hunk 3]
```diff
@@ -50,6 +51,67 @@ protected override async Task<SmugglerProgressBase.DatabaseRecordProgress> Proce
...
+                var typeMsg = string.IsNullOrEmpty(name) ? $"{type}" : $"{type} task '{name}'";
+                result.AddWarning($"Skipped {typeMsg} - it is currently not supported in Sharding");
+            }
+        }
+
+        protected override bool ShouldSkipIndex(IndexDefinitionAndType index, out string msg)
+        {
+            msg = null;
+
+            if (index.Type == IndexType.AutoMap || index.Type == IndexType.AutoMapReduce)
```
Comment (ppekrol) [inline src/Raven.Server/Smuggler/Documents/ShardedDatabaseSmuggler.cs:95]: use `index.Type.IsAuto()` next time :)

**Pull request merged**